dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue

```diff
@@ -140,30 +342,163 @@ export default defineComponent({
+      uploadFiles: [] as ExtendedCompanyReport[],
+      fiscalYearEnd: "" as Date | "",
+      convertedFiscalYearEnd: "",
+      reportingPeriod: new Date(),
+      assuranceData: [
+        { label: "None", value: "None" },
+        { label: "LimitedAssurance", value: "LimitedAssurance" },
+        { label: "ReasonableAssurance", value: "ReasonableAssurance" },
+      ],
+      maxFileSize: UPLOAD_MAX_FILE_SIZE_IN_BYTES,
+      euTaxonomyKPIsModel: euTaxonomyPseudoModelAndMappings,
+      euTaxonomyKpiNameMappings,
+      euTaxonomyKpiInfoMappings,
+      checkCustomInputs,
+      formatBytesUserFriendly,
+      route: useRoute(),
+      waitingForData: false,
+      editMode: false,
+
+      postEuTaxonomyDataForFinancialsProcessed: false,
+      messageCount: 0,
+      postEuTaxonomyDataForFinancialsResponse: null as AxiosResponse<DataMetaInformation> | null,
+      humanizeString: humanizeString,
+      onThisPageLinksStart: [
+        { label: "Upload company reports", value: "uploadReports" },
+        { label: "Basic information", value: "basicInformation" },
+        { label: "Assurance", value: "assurance" },
+        { label: "Add KPIs", value: "addKpis" },
+      ],
+      onThisPageLinks: [] as { label: string; value: string }[],
+      kpisModel: [
+        { label: "Credit Institution", value: "creditInstitutionKpis" },
+        { label: "Investment Firm", value: "investmentFirmKpis" },
+        { label: "Insurance & Re-insurance", value: "insuranceKpis" },
+        { label: "Asset Management", value: "assetManagementKpis" },
+      ],
+      selectedKPIs: [] as { label: string; value: string }[],
+      confirmedSelectedKPIs: [] as { label: string; value: string }[],
+      computedFinancialServicesTypes: [] as string[],
+      reportingPeriodYear: new Date().getFullYear(),
+    };
+  },
+  computed: {
+    namesOfFilesToUpload(): string[] {
+      const namesFromFilesToUpload = this.filesToUpload.map((el) => el.name.split(".")[0]);
+      const namesFromUploadedFiles = this.uploadFiles.map((el) => el.name.split(".")[0]);
+      return [...new Set([...namesFromFilesToUpload, ...namesFromUploadedFiles])];
     },
-    inputClass: {
-      "formkit-input": false,
-      "p-inputtext": true,
-      "w-full": true,
+  },
+  watch: {
+    confirmedSelectedKPIs: function (newValue: { label: string; value: string }[]) {
+      this.computedFinancialServicesTypes = newValue.map((el: { label: string; value: string }): string => {
+        return this.euTaxonomyKPIsModel.companyTypeToEligibilityKpis[
+          el.value as keyof typeof this.euTaxonomyKPIsModel.companyTypeToEligibilityKpis
+        ];
+      });
     },
-    postEuTaxonomyDataForFinancialsProcessed: false,
-    messageCount: 0,
-    formInputsModel: {},
-    postEuTaxonomyDataForFinancialsResponse: null,
-    humanizeString: humanizeString,
-  }),
+    reportingPeriod: function (newValue: Date) {
+      this.reportingPeriodYear = newValue.getFullYear();
+    },
```

Comment:
I think using a computed prop for this is a little cleaner.

So 
`computed: {
    reportingPeriodYear() {
      return this.reportingPeriod.getFullYear();
    },`

Reply:
ok